### PR TITLE
feat(monitor): add resource to manage Inhibition rules

### PR DESCRIPTION
**Pull request by dbonf**
* add resource to manage Inhibition rules
* add new field to attach labels to alerts

### CODE REVIEW

**Review by antonio-cuomo — APPROVED** [review]

File: website/docs/r/monitor_inhibition_rule.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+* `description` - (Optional) The description of the Inhibition Rule.
+
+* `enabled` - (Optional) Whether to enable the Inhibition Rule. Default: `true`.
+
+* `equal` - (Optional) List of labels that must have an equal value in the source and target alert for the inhibition to take effect.
+
+### `source_matchers`
+
+List of source matchers for which one or more alerts have to exist for the inhibition to take effect.
```
Comment (PanharithChhum) [inline website/docs/r/monitor_inhibition_rule.md:57]: This closely resembles the OSS docs description but I think we can aim for even more clarity. 

> Source Matchers define the label combinations that an alert occurrence must match for the inhibition rule to apply. When an alert occurrence satisfies this criteria, it activates the inhibition rule, which can then suppress notifications for target alerts.

File: website/docs/r/monitor_inhibition_rule.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+* `label_name`: (Required) Label to match.
+
+* `operator`: (Required) Match operator. It can be `EQUALS`, `NOT_EQUALS`, `REGEXP_MATCHES`, `NOT_REGEXP_MATCHES`.
+
+* `value`: (Required) Label value to match in case operator is of type equality, or a valid regular expression in case of operator is of type regex.
+
+### `target_matchers`
+
+List of target matchers that have to be fulfilled by the target alerts to be muted.
```
Comment (PanharithChhum) [inline website/docs/r/monitor_inhibition_rule.md:69]: Adapting the previous description for source matcher

> Target Matchers define the label combinations that an alert occurrence must match for the inhibition rule to suppress the alert notification. When a target alert occurrence meets these criteria, its notifications will be inhibited if the source matchers' conditions are also met.

File: website/docs/r/monitor_inhibition_rule.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+## Argument Reference
+
+* `name` - (Optional) The name of the Inhibition Rule. If provided, it must be unique.
+
+* `description` - (Optional) The description of the Inhibition Rule.
+
+* `enabled` - (Optional) Whether to enable the Inhibition Rule. Default: `true`.
+
+* `equal` - (Optional) List of labels that must have an equal value in the source and target alert for the inhibition to take effect.
```
Comment (PanharithChhum) [inline website/docs/r/monitor_inhibition_rule.md:53]: How about

> List of label names that must have identical values in both the source and target alert occurrences for the inhibition rule to apply

Comment (dbonf) [reply website/docs/r/monitor_inhibition_rule.md]: ok i added the suggestion.

File: website/docs/r/monitor_inhibition_rule.md (hunk 2 above)
Comment (dbonf) [reply website/docs/r/monitor_inhibition_rule.md]: ok i added the suggestion.

File: website/docs/r/monitor_inhibition_rule.md (hunk 1 above)
Comment (dbonf) [reply website/docs/r/monitor_inhibition_rule.md]: ok i added the suggestion.

**Review by michelevergnano — APPROVED** [review]

**Review by filiptubic — APPROVED** [review]

**Review by filiptubic — APPROVED** [review]

**Pull request merged**